File: backend/backend/Models/Group.cs
namespace backend.Models
{
    public class Group
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Category { get; set; }
        public Image GroupImage { get; set; }
        public Image GroupBackGroundImage { get; set; }
        public ICollection<GroupAdmin> Admins { get; set; } = new List<GroupAdmin>();
        public ICollection<GroupFollower> Followers { get; set; } = new List<GroupFollower>();
        public ICollection<GroupPost> Posts { get; set; } = new List<GroupPost>();
    }
}

File: client/src/app/models/Group.ts
import { Image } from "./Image"

export interface Group{
    id:string,
    name:string,
    description:string,
    category:string,
    isAdmin:boolean,
    followers: number,
    image:string,
    backgroundImage:string,
    follow:boolean
}

export interface GroupPost{
    id:string,
    content:string,
    group:Group,
    date:string,
    images: Image[],
    documents: Document[],
    likes:number,
    liked:boolean,
}


export interface GroupCreate{
    name:string,
    description:string,
    category:string,
    image?:File,
    backgroundImage?:File,
}

export interface GroupUpdate{
    id:string,
    name:string,
    description:string,
    category:string,
}